src/erlcloud_ddb_util.erl

```diff
@@ -309,22 +311,39 @@ q_all(Table, KeyConditionsOrExpression, Opts) ->
 %% @end
 %%------------------------------------------------------------------------------
 
--spec q_all(table_name(), conditions() | expression(), q_all_opts(), aws_config()) -> items_return().
+-spec q_all(table_name(),
+            conditions() | expression(),
+            q_all_opts(),
+            aws_config()) -> items_return().
 q_all(Table, KeyConditionsOrExpression, Opts, Config) ->
     q_all(Table, KeyConditionsOrExpression, Opts, Config, [], undefined).
 
--spec q_all(table_name(), conditions() | expression(), q_all_opts(), aws_config(), [[out_item()]], key() | undefined)
-           -> items_return().
-q_all(Table, KeyConditionsOrExpression, Opts, Config, Acc, StartKey) ->
-    case erlcloud_ddb2:q(Table, KeyConditionsOrExpression,
-                         [{exclusive_start_key, StartKey} | set_out_opt(Opts)], 
-                         Config) of
+-spec q_all(table_name(),
+            conditions() | expression(),
+            q_all_opts(),
+            aws_config(),
+            [[out_item()]],
+            key() | undefined) -> items_return().
+q_all(Tbl, KeyConditionsOrExpression, Opts0, Config, Acc, StartKey) ->
+    Opts = [{exclusive_start_key, StartKey}|set_out_opt(Opts0)],
+    case erlcloud_ddb2:q(Tbl, KeyConditionsOrExpression, Opts, Config) of
         {error, Reason} ->
             {error, Reason};
-        {ok, #ddb2_q{last_evaluated_key = undefined, items = Items}} ->
-            {ok, flatreverse([Items | Acc])};
-        {ok, #ddb2_q{last_evaluated_key = LastKey, items = Items}} ->
-            q_all(Table, KeyConditionsOrExpression, Opts, Config, [Items | Acc], LastKey)
+        {ok, #ddb2_q{last_evaluated_key = undefined,
+                     items              = undefined,
+                     count              = Count}} ->
+            {ok, lists:sum([Count|Acc])};
+        {ok, #ddb2_q{last_evaluated_key = undefined,
+                     items              = Items}} ->
+            {ok, flatreverse([Items|Acc])};
+        {ok, #ddb2_q{last_evaluated_key = LastKey,
+                     items              = Items,
+                     count              = Count}} ->
+            UpdAcc = case Items of
```

Comment:
Actually it looks better for me having one case clause for case when `last_evaluated_key /= undefined`. I think it makes sense to call `q_all(Tbl, KeyConditionsOrExpression, Opts, Config, UpdAcc, LastKey)` just once in one clause.